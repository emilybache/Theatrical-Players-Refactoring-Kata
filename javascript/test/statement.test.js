const statement = require("../src/statement");
const fs = require("fs");

test("example statement", () => {
  const invoice = JSON.parse(
    fs.readFileSync("test/mocks/invoice.json", "utf8")
  );
  const plays = JSON.parse(fs.readFileSync("test/mocks/plays.json", "utf8"));
  expect(statement(invoice, plays)).toMatchSnapshot();
});

test("statement with new play types", () => {
  const invoice = JSON.parse(
    fs.readFileSync("test/mocks/invoice_new_plays.json", "utf8")
  );
  const plays = JSON.parse(
    fs.readFileSync("test/mocks/new_plays.json", "utf8")
  );
  expect(() => {
    statement(invoice, plays);
  }).toThrow(/unknown type/);
});

test("statement with underperformed plays", () => {
  const invoice = JSON.parse(
    fs.readFileSync("test/mocks/invoice_less_audience.json", "utf8")
  );
  const plays = JSON.parse(fs.readFileSync("test/mocks/plays.json", "utf8"));
  expect(statement(invoice, plays)).toMatchSnapshot();
});
