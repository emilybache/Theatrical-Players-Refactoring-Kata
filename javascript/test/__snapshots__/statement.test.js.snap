// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example statement 1`] = `
"Statement for BigCo
 Hamlet: $650.00 (55 seats)
 As You Like It: $580.00 (35 seats)
 Othello: $500.00 (40 seats)
Amount owed is $1,730.00
You earned 47 credits
"
`;

exports[`statement with underperformed plays 1`] = `
"Statement for BigCo
 Hamlet: $400.00 (30 seats)
 As You Like It: $360.00 (20 seats)
 Othello: $400.00 (29 seats)
Amount owed is $1,160.00
You earned 4 credits
"
`;
